documentation/puppet-api/v3/environment_modules.markdown

```diff
@@ -0,0 +1,137 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Environment Modules"
+canonical: "/puppetserver/latest/puppet-api/v3/environment_modules.html"
+---
+
+[classes]: https://docs.puppet.com/puppet/latest/reference/lang_classes.html
+[deprecated WEBrick Puppet master]: https://docs.puppet.com/puppet/latest/reference/services_master_webrick.html
+[node definitions]: https://docs.puppet.com/puppet/latest/reference/lang_node_definitions.html
+[defined types]: https://docs.puppet.com/puppet/latest/reference/lang_defined_types.html
+[`environment_timeout`]: https://docs.puppet.com/puppet/latest/reference/config_file_environment.html#environmenttimeout
+[resource type API]: https://docs.puppet.com/puppet/latest/reference/http_api/http_resource_type.html
+[`manifest` setting]: https://docs.puppet.com/puppet/latest/reference/config_file_environment.html#manifest
+
+[`auth.conf`]: ../../config_file_auth.markdown
+[`puppetserver.conf`]: ../../config_file_puppetserver.markdown
+[environment cache API]: ../../admin-api/v1/environment-cache.markdown
+
+The environment modules API will return information about what modules are
+installed for the requested environment.
+
+This endpoint is available only when the Puppet master is running Puppet Server, not
+on Ruby Puppet masters, such as the [deprecated WEBrick Puppet master][]. It also ignores
+the Ruby-based Puppet master's authorization methods and configuration. See the
+[Authorization section](#authorization) for more information.
+
+## `GET /puppet/v3/environment_modules?environment=:environment`
+
+Making a request with no query parameters is not supported and returns an HTTP 400 (Bad
+Request) response.
+
+### Supported HTTP Methods
+
+GET
+
+### Supported Formats
+
+JSON
+
+### Query Parameters
+
+Provide one parameter to the GET request:
+
+* `environment`: Only the classes and parameter information pertaining to the specified
+environment will be returned for the call.
+
+### Responses
+
+#### GET request with results
+
+```
+GET /puppet/v3/environment_modules?environment=env
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "modules": [
+        {
+            "forge_name": "puppetlabs/ntp",
```

Comment:
This reminds me, I need to double check what happens when a module is installed that isn't form the forge. Not sure if the key is just missing from the `Puppet::Modules` object or if it's null.